src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -402,24 +394,24 @@ const emit = defineEmits<{
   (event: 'query-change', data: string): void
 }>()
 
-const query = ref(props.query)
-const isProcessingInternally = ref(false)
-const isRegExpMode = ref(false)
-const isFilterMode = ref(false)
+const query = ref<string>(props.query)
+const isProcessingInternally = ref<boolean>(false)
+const isRegExpMode = ref<boolean>(false)
+const isFilterMode = ref<boolean>(false)
 const regExpError = ref<Error | null>(null)
 const codeBlock = ref<HTMLElement | null>(null)
 const codeBlockSearchInput = ref<HTMLInputElement | null>(null)
 const codeBlockCopyButton = ref<typeof KButton | null>(null)
-const numberOfMatches = ref(0)
+const numberOfMatches = ref<number>(0)
 const matchingLineNumbers = ref<number[]>([])
 const currentLineIndex = ref<null | number>(null)
 
-const totalLines = computed(() => Array.from({ length: props.code.split('\n').length }, (_, index) => index + 1))
-const maxLineNumberWidth = computed(() => totalLines.value[totalLines.value.length - 1].toString().length + 'ch')
-const linePrefix = computed(() => props.id.toLowerCase().replace(/\s+/g, '-'))
-const isProcessing = computed(() => props.isProcessing || isProcessingInternally.value)
-const isShowingFilteredCode = computed(() => isFilterMode.value && filteredCode.value !== '')
-const filteredCode = computed(function() {
+const totalLines = computed<number[]>(() => Array.from({ length: props.code.split('\n').length }, (_, index) => index + 1))
+const maxLineNumberWidth = computed<string>(() => totalLines.value[totalLines.value.length - 1].toString().length + 'ch')
+const linePrefix = computed<string>(() => props.id.toLowerCase().replace(/\s+/g, '-'))
+const isProcessing = computed<boolean>(() => props.isProcessing || isProcessingInternally.value)
+const isShowingFilteredCode = computed<boolean>(() => isFilterMode.value && filteredCode.value !== '')
+const filteredCode = computed<string>(function() {
```

Comment:
For computed, to match the existing syntax, we define the type inside the method
```suggestion
const totalLines = computed((): number[] => Array.from({ length: props.code.split('\n').length }, (_, index) => index + 1))
const maxLineNumberWidth = computed((): string => totalLines.value[totalLines.value.length - 1].toString().length + 'ch')
const linePrefix = computed((): string => props.id.toLowerCase().replace(/\s+/g, '-'))
const isProcessing = computed((): boolean => props.isProcessing || isProcessingInternally.value)
const isShowingFilteredCode = computed((): boolean => isFilterMode.value && filteredCode.value !== '')
const filteredCode = computed((): string => {
```